core/trino-spi/src/main/java/io/trino/spi/type/Decimals.java

```diff
@@ -295,7 +327,7 @@ public static void writeShortDecimal(BlockBuilder blockBuilder, long value)
     public static long rescale(long value, int fromScale, int toScale)
     {
         if (toScale < fromScale) {
-            throw new IllegalArgumentException("target scale must be larger than source scale");
```

Comment:
let's keep current behaviour in `rescale` method. If somebody wants rounding, this needs to be explicit.

Reply:
Okay, will leave the original function as is and move the rescale to a different method.